prow/gerrit/adapter/adapter.go

```diff
@@ -253,9 +253,11 @@ func (c *Controller) processChange(latest client.LastSyncState, changeChan <-cha
 		now := time.Now()
 
 		result := client.ResultSuccess
-		if err := c.triggerJobs(log, instance, change); err != nil {
-			result = client.ResultError
-			log.WithError(err).Info("Failed to trigger jobs based on change")
+		if !c.shouldSkipProcessingChange(instance, change) {
+			if err := c.triggerJobs(log, instance, change); err != nil {
+				result = client.ResultError
+				log.WithError(err).Info("Failed to trigger jobs based on change")
+			}
 		}
```

Comment:
nit: When we have conditionals that might cause us to exit the function (or really any code block `{ ... }`. It is typically best practice to invert the conditional as needed so that the conditional is entered in the case where execution would be interrupted. This prevents the uninterrupted execution path from becoming nested/indented which improves readability.
```suggestion
		if c.shouldSkipProcessingChange(instance, change) {
		    return
		}
		if err := c.triggerJobs(log, instance, change); err != nil {
			result = client.ResultError
			log.WithError(err).Info("Failed to trigger jobs based on change")
		}
```

Reply:
Agreed with avoiding nested if statements, however doing that means skipping the following metrics collection and update the `lastest` map.

Reply:
Ah ok, that makes sense to me then.

One thing I just thought to add though is that it would probably be handy to log a debug level message (with appropriate log fields) whenever we successfully skip processing of a change.  Otherwise that could be a bit tricky to debug from the logs.

Reply:
Added a log message.